### PR TITLE
Linux support for message passing

### PR DESCRIPTION
**Pull request by funnbot**
Response to #2 
Re implements bridge message passing with unix sockets, a bit more complicated than it should be due to being non-blocking, and it may have been a better idea to use boost.asio, as that would allow the switch to websockets to be seamless..
I tested the best I could with compiler explorer, surprisingly local sockets work there... https://godbolt.org/z/G9sY5K4da
I set the socket path is `/tmp/SlimeVRDriver` as it doesn't require root to write to.

This pr also fixes cmake builds for linux, adds ubuntu-latest to the github action matrix, and uploads the artifact.
Once someone begins work on the Java side, they can download it and hopefully be able to test them together.

I'm funnbot#8284 on the slimevr discord if you have questions.

### CODE REVIEW

**Comment by kitlith** [comment]
Is there anything available to test against this build of the driver, such as a corresponding PR to slimevr-server?

**Comment by kitlith** [comment]
yeah, i missed the "Once someone begins work on the Java side", my bad.

**Comment by Eirenliel** [comment]
Please rebase to main and fix confclits. Also we need testing with https://github.com/SlimeVR/SlimeVR-Server/pull/273

**Review by ImUrX — CHANGES_REQUESTED** [review]
it seems the CMake script generates an ``SlimeVR`` folder instead of using the existing ``slimevr`` one. Surely this bug wasn't noticed before because NTFS isnt case-sensitive, but most linux filesystems are by default

File: src/bridge/unix-sockets.hpp [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+        return mAddress.sun_family == sFamily; // not used with wrong socket type
+    }
+private:
+    socklen_t mSize;
+    sockaddr_un_t mAddress{};
+};
+
+class LocalSocket : public Socket {
+    static constexpr int sDomain = AF_UNIX, // unix domain socket
+                         sType = SOCK_SEQPACKET, // message boundaries and connection oriented
```
Comment (ImUrX) [inline src/bridge/unix-sockets.hpp:250]: Change to SOCK_STREAM

**Review by ImUrX — CHANGES_REQUESTED** [review]

**Comment by ImUrX** [comment]
now there are more errors, it wont compile for linux because of the ``#ELIF`` in the openvr.cpp

will wait for fix and try after :3

**Review by ImUrX — APPROVED** [review]

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**